### PR TITLE
[ci] check JavaScript code with `biome` tool

### PR DESCRIPTION
**Pull request by StrikerRUS**
`biome` project page: https://biomejs.dev/.

I found it during reading the following article: https://blog.logrocket.com/eslint-adoption-guide/.

The most popular ESLint was too hard to configure (especially after the recent breaking changes: https://github.com/eslint/eslint/issues/18391).
> Over the years, many popular and widespread shared configurations have been developed. However, with every breaking change in ESLint, the community projects need to migrate.

Some other alternatives like [`standard`](https://github.com/standard/standard), [`neostandard`](https://github.com/neostandard/neostandard) and [`jslint`](https://www.jslint.com/) were too aggressive with their feature of absence of configuration.

So I think `biome` as all-in-one tool (linter+formatter) outside of `nodejs` environment with easy configuration for non-frontenders is a good choice for LightGBM project.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
 insert_final_newline = true
+end_of_line = lf
 indent_style = space
 indent_size = 2
 
-[*.{py,sh,js,ps1}]
+[*.{py,sh,ps1,js,json,css}]
 indent_size = 4
-line_length = 120
+max_line_length = 120
```
Comment (StrikerRUS) [inline .editorconfig:13]: https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#max_line_length

File: .editorconfig [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
 skip = external_libs
 known_first_party = lightgbm
 
-# Placeholder files
-[{*.gitkeep,__init__.py}]
-insert_final_newline = none
```
Comment (StrikerRUS) [inline .editorconfig:18]: Unfortunately, `biome` fails to read the config here. Also, [`insert_final_newline=unset|off`](https://github.com/biomejs/biome/pull/3907) doesn't help.

```
configuration ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  × Failed to parse the .editorconfig file.
    
    Caused by:
      Custom("expected 'true' or 'false'")
```
Anyway, [`none` is not allowed value in editorconfig](https://spec.editorconfig.org/), so these lines haven't actually worked.

File: .ci/test.sh [hunk 3]
```diff
@@ -113,12 +114,14 @@ if [[ $TASK == "lint" ]]; then
         'r-lintr>=3.1.2'
     # shellcheck disable=SC1091
     source activate "${CONDA_ENV}"
-    echo "Linting Python code"
-    bash ./.ci/lint-python.sh || exit 1
+    echo "Linting Python and bash code"
+    bash ./.ci/lint-python-bash.sh || exit 1
```
Comment (StrikerRUS) [inline .ci/test.sh:118]: Due to [recently added `shellcheck`](https://github.com/microsoft/LightGBM/pull/6685) in pre-commit hook.

File: biome.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (StrikerRUS) [inline biome.json:1]: https://biomejs.dev/reference/configuration/

File: .editorconfig [hunk 5]
```diff
@@ -1,22 +1,19 @@
...
-charset=utf-8
+charset = utf-8
 trim_trailing_whitespace = true
 insert_final_newline = true
+end_of_line = lf
 indent_style = space
 indent_size = 2
 
-[*.{py,sh,js,ps1}]
+[*.{py,sh,ps1,js,json,css}]
```
Comment (jameslamb) [inline .editorconfig:11]:
```suggestion
[*.{py,sh,ps1,js,json}]
```

There aren't any `.css` files checked into source control here.

```shell
git ls-files | grep css
```

But there can sometimes be some laying around in your local development environment after rendering the sphinx docs or pkgdown docs.

Given that they're auto-generated... I don't think we want to generate any linter warnings for such files.

**Review by jameslamb — APPROVED** [review]
Interesting! I'm so used to `prettier` in javascript projects, I'd never really tried anything else. `biome` looks interesting, and I'm glad we can install it from conda-forge.

I'm open to trying this, even though we only have about 70 lines of Javacsript in this repo... seems cheap to install and like it runs quickly. Thanks!

**Comment by StrikerRUS** [comment]
@jameslamb 
> and I'm glad we can install it from conda-forge.

Yeah! I was really excited that we can avoid bringing nodejs ecosystem here for just one js-file.

File: .editorconfig (hunk 5 above)
Comment (StrikerRUS) [reply .editorconfig:11]: I added `css` for possible **future** css files in our repo. But I didn't think about the case you described. I think it makes more sense than mine.

**Pull request merged**